crates/ruff_python_semantic/src/context.rs

```diff
@@ -708,3 +758,17 @@ pub struct Snapshot {
     definition_id: DefinitionId,
     flags: ContextFlags,
 }
+
+#[derive(Debug)]
+pub enum ResolvedReference {
+    /// The reference is resolved to a specific binding.
```

Comment:
What's the reason that we need to store the `ScopeId` and `BindingId`. Is there no way for us to do a `BindingId` -> `Binding` lookup?

Reply:
It's because we have this weird behavior at the level above (which should be moved into this method, I think) whereby we check if there are other names in the scope that need to be marked as "used". This is mainly for submodule imports, e.g., if you have:

```py
import pyarrow as pa
import pyarrow.csv
print(pa.csv.read_csv("test.csv"))
```

Then we end up returning `pa` as the resolved reference, but we actually want to mark `pyarrow.csv` as used too.